ipatests/create_external_ca.py

```diff
@@ -45,7 +51,7 @@ def create_ca(self, cn=ISSUER_CN):
         """
         self.ca_key = rsa.generate_private_key(
             public_exponent=65537,
-            key_size=2048,
+            key_size=int(ipalib.constants.PKI_CA_SIGNING_KEY_SIZE),
```

Comment:
The ``int`` call is not necessary.

Reply:
It *is* necessary:
```
# python3 ./create_external_ca.py --csr /var/lib/ipa/ca.csr
Traceback (most recent call last):
  File "./create_external_ca.py", line 210, in <module>
    main()
  File "./create_external_ca.py", line 196, in main
    root_ca = external_ca.create_ca()
  File "./create_external_ca.py", line 55, in create_ca
    backend=default_backend(),
  File "/usr/lib64/python3.6/site-packages/cryptography/hazmat/primitives/asymmetric/rsa.py", line 118, in generate_private_key
    _verify_rsa_parameters(public_exponent, key_size)
  File "/usr/lib64/python3.6/site-packages/cryptography/hazmat/primitives/asymmetric/rsa.py", line 129, in _verify_rsa_parameters
    if key_size < 512:
TypeError: '<' not supported between instances of 'str' and 'int'

```

Reply:
You have to define ``PKI_CA_SIGNING_KEY_SIZE`` as number, too.

Reply:
It is really a conflict between RawConfigParser.set() and python cryptography code. Either I should do `str(foo)` in `config.set()` or `int(foo)` in these calls.